harpy/report/plots.py

```
@@ -0,0 +1,233 @@
+        alt.renderers.enable(previous)
+
+def piechart(df: pd.DataFrame, title: str = "", goodval: str|None = None, lgtitle: bool = False, lgpos: str = "bottom"):
+    total = df['count'].sum()
+    _names = list(df.columns)
+    _n = set(df[_names[0]])
+    _color = alt.Color(
+        f'{_names[0]}:N', title = _names[0].title(),
+        legend=alt.Legend(orient=lgpos, title = None) if not lgtitle else alt.Legend(orient=lgpos)
+        )
+    if goodval:
+        badval = [i for i in _n if i != goodval][0]
```

Ruff (0.15.14)

[warning] Prefer `next(i for i in _n if i != goodval)` over single element slice

Replace with `next(i for i in _n if i != goodval)`

(RUF015)